src/modules/Builder/features/Activities/Activities.tsx

```diff
@@ -75,8 +76,25 @@ export const Activities = () => {
         {activities?.length ? (
           activities.map((item: ActivityFormValues, index: number) => {
             const handleEdit = () => navigateToActivity(item.key ?? item.id);
-            //TODO: check if dependent activity flow is required to remove
-            const handleRemove = () => removeActivity(index);
+
+            const handleRemove = () => {
```

Comment:
can y move this outside map? y have (n^3) complexity

Reply:
done